manager/integration/tests/test_cluster_autoscaler.py

```diff
@@ -0,0 +1,389 @@
+    And Create deployment with cpu request.
+
+    When Trigger CA to scale-up by increase deployment replicas.
+         (double the node number, not including host node)
+    Then Cluster should have double the node number.
+
+    When Trigger CA to scale-down by decrease deployment replicas.
+    Then Cluster should scale-down to original node number.
+    """
+    # Cleanup
+    def finalizer():
+        configure_node_scale_down(core_api, client.list_node(), disable="")
+        set_node_cordon(core_api, host_id, False)
+    request.addfinalizer(finalizer)
+
+    host_id = get_self_host_id()
+    host_node = client.by_id_node(host_id)
+    configure_node_scale_down(core_api, [host_node], disable="true")
+    set_node_cordon(core_api, host_id, True)
+
+    update_setting(client, SETTING_K8S_CLUSTER_AUTOSCALER_ENABLED, "true")
+
+    nodes = client.list_node()
+    scale_size = len(nodes)-1
+
+    scale_up_replica = get_replica_count_to_scale_up(
+        core_api, scale_size, CPU_REQUEST
+    )
+
+    deployment_name = "ca-scaling-control"
+    deployment = make_deployment_cpu_request(deployment_name, CPU_REQUEST)
+    create_and_wait_deployment(apps_api, deployment)
+
+    deployment["spec"]["replicas"] = scale_up_replica
+    apps_api.patch_namespaced_deployment(body=deployment,
+                                         namespace='default',
+                                         name=deployment_name)
+
+    wait_cluster_autoscale_up(client, nodes, scale_size)
+
+    scale_down_replica = 0
+    deployment["spec"]["replicas"] = scale_down_replica
+    apps_api.patch_namespaced_deployment(body=deployment,
+                                         namespace='default',
+                                         name=deployment_name)
+    nodes = client.list_node()
+    client = wait_cluster_autoscale_down(client, core_api, nodes, scale_size)
+
+
+@pytest.mark.skip(reason="require K8s cluster-autoscaler")
+@pytest.mark.cluster_autoscaler  # NOQA
+def test_cluster_autoscaler_all_nodes_with_volume_replicas(client, core_api, apps_api, make_deployment_cpu_request, volume_name, pod_make, request):  # NOQA
+    """
+    Scenario: Test CA scale down all nodes with volume replicas
+
+    Given Cluster with Kubernetes cluster-autoscaler.
+    And Longhorn installed.
+    And Set `kubernetes-cluster-autoscaler-enabled` to `true`.
+    And Create volume.
+    And Attach the volume.
+    And Write some data to volume.
+    And Detach the volume.
+    And Create deployment with cpu request.
+
+    When Trigger CA to scale-up by increase deployment replicas.
+         (double the node number, not including host node)
+    Then Cluster should have double the node number.
+
+    When Annotate new nodes with
+         `cluster-autoscaler.kubernetes.io/scale-down-disabled`.
+         (this ensures scale-down only the old nodes)
+    And Trigger CA to scale-down by decrease deployment replicas.
+    Then Cluster should have original node number + 1 blocked node.
```

Comment:
I prefer not to touch/scale down the old nodes. The backup target pods are running on those old nodes.